core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -1059,6 +1057,31 @@ private void unparseOtherFunction(SqlWriter writer, SqlCall call, int leftPrec,
     }
   }
 
+  protected void unParseDateTime(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
+    String dateFormat = call.operand(0) instanceof SqlCharStringLiteral
+        ? ((NlsString) requireNonNull(((SqlCharStringLiteral) call.operand(0)).getValue()))
+        .getValue() : call.operand(0).toString();
+    SqlCall formatCall = null;
+    switch (call.getOperator().getName()) {
+    case "PARSE_DATE":
+      formatCall = PARSE_DATE.createCall(SqlParserPos.ZERO,
+          createDateTimeFormatSqlCharLiteral(dateFormat), call.operand(1));
+      break;
+    case "PARSE_TIME":
+      formatCall = PARSE_TIME.createCall(SqlParserPos.ZERO,
+          createDateTimeFormatSqlCharLiteral(dateFormat), call.operand(1));
+      break;
+    case "PARSE_DATETIME":
+    case "PARSE_TIMESTAMP":
+      formatCall = PARSE_DATETIME.createCall(SqlParserPos.ZERO,
+          createDateTimeFormatSqlCharLiteral(dateFormat), call.operand(1));
+      break;
+    default:
+      unparseOtherFunction(writer, call, leftPrec, rightPrec);
+    }
```

Comment:
switch is not required as code is getting repeated.

SqlCall formatCall = call.getOperator().createCall(SqlParserPos.ZERO,
        createDateTimeFormatSqlCharLiteral(dateFormat), call.operand(1));

and default is not required. If requuired please mention here and do changes accordingly.

Reply:
switch is not required as mentioned earlier you don't have any other case except parse_date and parse_time